internal/transactionsigner/transactionsigner.go

```diff
@@ -0,0 +1,110 @@
+package transactionsigner
+
+import (
+	"fmt"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	ec "github.com/bitcoin-sv/go-sdk/primitives/ec"
+	"github.com/bitcoin-sv/go-sdk/script"
+	trx "github.com/bitcoin-sv/go-sdk/transaction"
+	sighash "github.com/bitcoin-sv/go-sdk/transaction/sighash"
+	"github.com/bitcoin-sv/go-sdk/transaction/template/p2pkh"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+)
+
+type TransactionSigner struct {
+	xPriv *bip32.ExtendedKey
+}
+
+func New(xPriv *bip32.ExtendedKey) *TransactionSigner {
+	return &TransactionSigner{
+		xPriv: xPriv,
+	}
```

Comment:
It looks like a nil check is missing, which could potentially lead to a panic during the `FinalizeTransaction` call. 😢